jdbc/src/main/java/camp/nextstep/jdbc/core/JdbcTemplate.java

```diff
@@ -23,7 +24,7 @@ public JdbcTemplate(final DataSource dataSource) {
     }
 
     public void update(final String sql, Object... args) {
-        try (final Connection connection = dataSource.getConnection();
+        try (final Connection connection = DataSourceUtils.getConnection(dataSource);
              final PreparedStatement pstmt = connection.prepareStatement(sql)) {
```

Comment:
```suggestion
        try {
            final Connection connection = DataSourceUtils.getConnection(dataSource);
            final PreparedStatement pstmt = connection.prepareStatement(sql);
```

> 미션을 구현했는데 UserServiceTest에서 testChangePassword 테스트에서 실패를 하고 있습니다.
디버깅을 했을 때 AppUserService에서 userHistoryDao.log(new UserHistory(user, createBy)); 를 실행할 때 jdbcTemplate클래스의 update메소드에서 `PreparedStatement를 할당할 때 에러가 발생하더라구요.

제가 로그상 확인한 에러는 다음과 같습니다.
`org.h2.jdbc.JdbcSQLNonTransientException: The object is already closed`

이미 closed 된 자원을 사용하고 있다고 로그가 말합니다.
그리고 미션에서 closed 된 자원이란 곧 connection 을 뜻합니다.

Try-With-Resources 문법에 대해서 더 깊게 공부해보면 좋을 것 같습니다.
지금과 같은 상황에서는 update 메서드를 호출할 때마다, connection 이 바로 closed 되게 되어,
같은 트랜잭션 안에 있는 다른 update 에서 connection 을 재활용하지 못합니다.

같은 트랜잭션이라면 connection 이 closed 되지 않고, 같은 connection 을 재사용할 수 있고,
트랜잭션이 완료되었을 때 connection 이 closed 되도록 해주세요.

참고로 맨 위의 suggestion code 는 이렇게 하면 테스트를 통과할 수 있다이지,
이렇게 고쳐야 한다는 뜻이 절대 아닙니다.
위의 코드에서는 그래서 connection 을 어디서 어떻게 close 할 것인지에 대한 부분이 빠져있는데,
이 부분을 구현해주시기 바랍니다.